lost-and-found-service-service/src/main/java/cz/muni/fi/service/ItemServiceImpl.java

```diff
@@ -24,55 +25,63 @@
     @Autowired
     private ItemDao itemDao;
 
+    @Autowired
+    CategoryService categoryService;
+
+    @Autowired
+    UserService userService;
+
+    @Autowired
+    LocationService locationService;
+
     @Override
-    public Item archiveItem(Item item) throws ServiceException{
+    public void archiveItem(Item item) throws ServiceException {
         if (item == null) {
-            throw new IllegalArgumentException("Item cannot be null");
+            throw new ServiceException("Item cannot be null");
```

Comment:
exceptions v tychto "advanced metodach" povazujem ako jedine za diskutabilne.

Inak by podla mna service mal proste volat dao, a neriesit veci ktore uz riesi dao

Reply:
suhlas, ze checky na veci co checkuje dao netreba, ale pri archivovani potrebujem checknut v akom je stave item a ak je nahodou null, nechcem aby mi to vyfailovalo na null dereferencii, tak si myslim, ze tam v tomto pripade check na null dava zmysel.

Reply:
tie checky som tam nechal len som zmenil aky exception hadzu